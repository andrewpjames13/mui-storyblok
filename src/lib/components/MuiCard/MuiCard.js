import React, { createElement } from 'react';
import PropTypes from 'prop-types';
import Card from '@material-ui/core/Grid';
import Storyblok from '../../utils/Storyblok';

import MuiCardActions from './components/MuiCardActions/MuiCardActions';
import MuiCardContent from './components/MuiCardContent/MuiCardContent';
import MuiCardHeader from './components/MuiCardHeader/MuiCardHeader';
import MuiCardMedia from './components/MuiCardMedia/MuiCardMedia';
import MuiVideo from '../MuiVideo/MuiVideo';

const MuiCard = ({
  rootClass,
  raised,
  content,
  height,
  width,
}) => {
  const components = {
    MuiCardActions,
    MuiCardContent,
    MuiCardHeader,
    MuiCardMedia,
    MuiVideo,
  };

  const styles = Storyblok.arrayToMuiStyles(rootClass, { height, width });

  return (
    <Card
      className={styles.root}
      raised={raised.toString()}
    >
      {content.map((item, index) => createElement(
        components[item.component],
        Object.assign(item, { key: index }),
      ))}
    </Card>
  );
};

export default MuiCard;

MuiCard.propTypes = {
  /**
   * stroyblok multiselect of css classes
   * Mui Override or extend the styles applied to the component.
   */
  rootClass: PropTypes.arrayOf(PropTypes.string),
  // mui prop: 'row' | 'row-reverse' | 'column' | 'column-reverse'
  /**
   * mui prop: true | false
   * If true, the card will use raised styling.
   */
  raised: PropTypes.bool,
  /**
   * Height of the card.
   */
  height: PropTypes.string,
  /**
   * Width of the card.
   */
  width: PropTypes.string,

  /**
   * Content passed to render
   * components:  MuiCardActions,
    MuiCardContent,
    MuiCardHeader,
    MuiCardMedia,
   */
  content: PropTypes.arrayOf(PropTypes.shape({
    component: PropTypes.string.isRequired,
  })).isRequired,
};

MuiCard.defaultProps = {
  raised: false,
  rootClass: [],
  height: '100%',
  width: '100%',
};
