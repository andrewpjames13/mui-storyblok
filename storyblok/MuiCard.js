module.exports = {
  "name": "MuiCard",
  "display_name": null,
  "updated_at": "2020-06-10T16:58:24.740Z",
  "schema": {
    "rootClass": {
      "type": "options",
      "pos": 0,
      "min_options": "",
      "max_options": "",
      "options": [
        {
          "value": "box-shadow: 5px 10px #FFF",
          "name": "box-shadow: 5px 10px;"
        },
        {
          "value": "background-color: #000",
          "name": "background-color black"
        },
        {
          "value": "color: #000",
          "name": "color-black"
        },
        {
          "value": "border-style: solid",
          "name": "border"
        },
        {
          "value": "max-width: 500",
          "name": "maxWidth"
        },
        {
          "value": "display: flex",
          "name": "flex"
        },
        {
          "value": "justify-content: center",
          "name": "justify center"
        }
      ]
    },
    "raised": {
      "type": "boolean",
      "description": "If true, the card will use raised styling.",
      "pos": 1
    },
    "content": {
      "type": "bloks",
      "restrict_components": true,
      "component_whitelist": [
        "MuiCardMedia",
        "MuiCardHeader",
        "MuiCardContent",
        "MuiCardActions",
        "MuiVideo"
      ],
      "pos": 2
    },
    "height": {
      "type": "text",
      "default_value": "100%",
      "description": "Height of the card"
    },
    "width": {
      "type": "text",
      "description": "Width of card",
      "default_value": "100%"
    }
  },
  "image": null,
  "preview_field": null,
  "is_root": false,
  "preview_tmpl": null,
  "is_nestable": true,
  "all_presets": [],
  "preset_id": null,
  "real_name": "MuiCard",
  "component_group_uuid": null
}