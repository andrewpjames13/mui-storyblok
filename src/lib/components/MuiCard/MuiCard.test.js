import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import MuiCard from './MuiCard';

function setup() {
  const props = {
    content: [{
      component: 'MuiCardActions',
      content: [{
        component: 'MuiButtonRedirect',
        redirectRoute: 'page-welcome',
        button: [{
          component: 'MuiButton',
          buttonText: 'buttonText',
        }],
      }],
    }, {
      component: 'MuiCardContent',
      content: [{
        component: 'MuiTypography',
        content: [{
          component: 'MuiText',
          text: 'text',
        }],
      }],
    }, {
      component: 'MuiCardHeader',
      title: 'title',
    }, {
      component: 'MuiCardMedia',
      alt: 'alt',
      title: 'title',
      image: 'img.com',
    }],
  };
  const comp = shallow(<MuiCard {...props} />);
  return { comp, props };
}

describe('<MuiCard />', () => {
  it('renders MuiCard', () => {
    const { comp } = setup();
    expect(comp).toBeDefined();
  });

  test('snapshot', () => {
    const { props } = setup();

    const tree = renderer.create(<MuiCard {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
