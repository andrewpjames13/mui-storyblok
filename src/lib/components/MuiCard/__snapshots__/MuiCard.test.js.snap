// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MuiCard /> snapshot 1`] = `
<div
  className="MuiGrid-root makeStyles-root-3"
  raised="false"
>
  <div
    className="MuiCardActions-root makeStyles-root-107 MuiCardActions-spacing"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-root-110 MuiButton-textSecondary MuiButton-fullWidth"
      data-testid="muiButton"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        buttonText
      </span>
    </button>
  </div>
  <div
    className="MuiCardContent-root makeStyles-root-143"
  >
    <p
      className="MuiTypography-root makeStyles-root-145 MuiTypography-body1"
    >
      text
    </p>
  </div>
  <div
    className="MuiCardHeader-root makeStyles-root-176"
  >
    <div
      className="MuiCardHeader-avatar"
    />
    <div
      className="MuiCardHeader-content"
    >
      <span
        className="MuiTypography-root MuiCardHeader-title MuiTypography-body2 MuiTypography-displayBlock"
      >
        title
      </span>
      <span
        className="MuiTypography-root MuiCardHeader-subheader MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
      >
        
      </span>
    </div>
    <div
      className="MuiCardHeader-action"
    />
  </div>
  <div
    alt="alt"
    className="MuiCardMedia-root makeStyles-root-183"
    style={
      Object {
        "backgroundImage": "url(\\"img.com\\")",
      }
    }
    title="title"
  />
</div>
`;
